core/src/connection/error.rs

```diff
@@ -84,14 +84,33 @@ pub enum PendingConnectionError<TTransErr> {
     Aborted,
 
     /// The peer identity obtained on the connection did not
-    /// match the one that was expected or is otherwise invalid.
-    InvalidPeerId,
+    /// match the one that was expected or is the local one.
+    WrongPeerId {
+        obtained: PeerId,
+        address: Multiaddr,
```

Comment:
```suggestion
        address: ConnectedPoint,
```
How about a `ConnectedPoint` instead of a plain `Multiaddr` here? Mostly for the sake of additional context.